website/src/components/Layout.tsx

```diff
@@ -0,0 +1,18 @@
+// https://nextjs.org/docs/basic-features/layouts
+
+import type { NextPage } from "next";
+
+import { Footer } from "./Footer";
+import { Header } from "./Header";
+
+export type NextPageWithLayout<P = {}, IP = P> = NextPage<P, IP> & {
+  getLayout?: (page: React.ReactElement) => React.ReactNode;
+};
+
+export const getLayout = (page: React.ReactElement) => (
```

Comment:
This component is pretty magical.  Can you add some high level comments on how it works.  My understanding is that any new page component should set a `getLayout` field and use this method.  That'll use this one unified layout.

Should we name this as something a little more specific like `getDefaultLayout` in the situation where we have another layout?

Reply:
You are totally right, I changed it now so it is the default layout (you don't have to specify anything in the components), with the option of overriding the layout if needed.